help_to_heat/frontdoor/routing/backwards_routing.py

```diff
@@ -14,20 +15,15 @@
 
 
 def get_prev_page(current_page, answers):
-    journey_page = start_page
-    journey = deque([govuk_start_page])
-
-    while len(journey) < max_journey_length:
-        if journey_page == current_page:
-            return journey.pop()
-
-        if journey_page == unknown_page:
-            return unknown_page
-
-        next_page = get_next_page(journey_page, answers)
-
-        journey.append(journey_page)
-
-        journey_page = next_page
-
-    return unknown_page
+    # there is no previous page to this
+    if current_page == govuk_start_page:
+        return unknown_page
+
+    try:
+        journey = calculate_journey(answers, current_page)
+
+        # if the page has been found, the last item in the list (index -1) will be the current page
+        # so, the item before the last one (index -2) will be the previous page
+        return journey[-2]
+    except CouldNotCalculateJourneyException:
+        return unknown_page
```

Comment:
I'm confused where we are with this: thought either this was being removed to just throw and bubble up, or we were logging here?

In which case is there any helpful message we should include with the error, for diagnostic purposes?

Reply:
yeah I'm a little undecided here on returning a telling value that there was an error vs. throwing. the reason I've gone this way is
- `get_next_page` doesn't throw an error if it can't find the next page so it feels more natural to follow suit
- there are times where we don't want this to throw. on the manual address pages (back again!) as we're hard coding a link to them, when they're loaded the back links can't be automatically generated. I added some code to the view to override this but if this function were to throw then the page would error before the override logic runs.
if we do want to preserve the error throwing behaviour I could add some exceptions to the routing logic to ensure in the case of the error manual pages it doesn't throw

Reply:
I agree, I've made it so this throws and the places where this can be ignored are in the schema, so in the cases where we know the back url the function isn't called